### DISCUSSION TITLE
I have build a Filecoin FEVM metadata extraction system, but the the EVM RPC QPS is very slow.

### DISCUSSION THREAD

**Discussion by cxytz01** [filecoin-project/community#623] [Q&A]
Here is my EVM transaction's receipt data extraction code: https://github.com/Spacescope/meta-task/blob/main/pkg/tasks/evmtask/receipt.go#L26

It takes at least 6 minutes, even 30 minutes to consume a Tipset.  Is there any room for optimization or an alternative solution?

The EVM RPC is very very slow, it can't be acceptable. Can I use Filecoin RPC instead?

**Comment by longfeiWan9** [comment]
@cxytz01 you did not use Json RPC calls to query tx receipt data, right? I do not think RPC calls are that slow. Lotus also offers eth compatible apis, you can check out

- [API source code ](https://github.com/filecoin-project/lotus/blob/master/api/api_full.go#L806)
- [eth API docs](https://github.com/filecoin-project/lotus/blob/master/documentation/en/api-v1-unstable-methods.md#Eth)
